app/services/dual-output/dual-output.ts

```diff
@@ -360,80 +381,291 @@ export class DualOutputService extends PersistentStatefulService<IDualOutputServ
   }
 
   /**
-   * Create or confirm nodes for vertical output when toggling vertical display
-   * @param sceneId - Id of the scene to map
+   * Create or confirm nodes for vertical output
    */
-  confirmOrCreateVerticalNodes(sceneId: string) {
-    this.convertSceneSources(sceneId);
-    if (!this.views.hasNodeMap(sceneId) && this.state.dualOutputMode) {
-      try {
-        this.createSceneNodes(sceneId);
-      } catch (error: unknown) {
-        console.error('Error toggling Dual Output mode: ', error);
-      }
-    } else {
-      try {
-        this.confirmOrAssignSceneNodes(sceneId);
-      } catch (error: unknown) {
-        console.error('Error toggling Dual Output mode: ', error);
-      }
+  confirmOrCreateDualOutputCollection() {
+    if (this.views?.sceneNodeMaps) {
+      this.validateDualOutputCollection();
+    } else if (this.state.dualOutputMode) {
+      this.convertSingleOutputToDualOutputCollection();
     }
+    if (this.state.isLoading) this.SET_IS_LOADING(false);
+  }
+
+  validateDualOutputCollection() {
+    this.SET_IS_LOADING(true);
+    // confirm dual output scene collection
+    this.scenesService.views.scenes.forEach(scene => {
+      if (this.views.sceneNodeMaps[scene.id]) {
+        this.confirmSceneNodeMap(scene.id, this.views.sceneNodeMaps[scene.id]);
+      } else {
+        this.createVerticalNodes(scene.id);
+      }
+    });
+    this.SET_IS_LOADING(false);
   }
 
-  convertSceneSources(sceneId: string) {
-    const sceneSources = this.scenesService.views.sceneSourcesForScene(sceneId);
-    if (sceneSources.length > 0) {
-      sceneSources.forEach(scene => this.confirmOrCreateVerticalNodes(scene.sourceId));
+  convertSingleOutputToDualOutputCollection() {
+    // if in dual output mode, convert the single output collection to dual output
+    this.scenesService.views.scenes.forEach(scene => {
+      this.createVerticalNodes(scene.id);
+    });
+  }
+
+  /**
+   * Create a vertical node to partner with the vertical node
+   * @param horizontalNode - Node to copy to the vertical display
+   *
+   * @remark The horizontal node id is always the key in the scene node map.
+   * The node map entry is so that the horizontal and vertical nodes can refer to each other.
+   */
+  createVerticalNode(horizontalNode: TSceneNode, verticalNodeId?: string): TSceneNode {
+    const scene = horizontalNode.getScene();
+
+    if (horizontalNode.isFolder()) {
+      // add folder and create node map entry
+      const folder = scene.createFolder(horizontalNode.name, {
+        id: verticalNodeId,
```

Comment:
im not sure how this works given that `verticalNodeId` is an optional parameter? how do we ensure that the folder has a valid id?